fix(llm): improve error detection and unify model selection

psyray opened the pull request:
fix #12 

Added more check for missing datas, like GPT API key, Ollama model empty ...
Now a popup is displayed when some datas are missing to complete LLM Analysis
<img width="811" height="343" alt="image" src="https://github.com/user-attachments/assets/dd9a1fc1-1460-4079-bb35-c18e208b8c10" />

- Add OpenAI API key redirects to API Vault
- Choose Ollama Models display the model selection
<img width="1105" height="719" alt="image" src="https://github.com/user-attachments/assets/769d27aa-333a-49e5-81da-d1e96982f190" />
- Cancel close the popup

LLM Analysis has been unified between Attack Surface and Vulnerability
No more duplicated code, only on popup for all

If an analysis is empty (main subject of the fix for #12) an analysis is launched again to prevent empty analysis to be loaded from cache (database)

psyray committed:
feat(llm): improve LLM config handling and model selection UX

- Added preflight checks for LLM configuration in the backend, returning structured errors when configuration is incomplete.
- Enhanced frontend to handle LLM configuration errors, prompting users to add API keys or select models as needed.
- Introduced new dialogs for LLM model selection and configuration, improving user experience for both attack surface and vulnerability detail flows.
- Improved robustness in markdown conversion and OpenAI parameter handling.

psyray committed:
fix(llm): improve handling of empty LLM vulnerability and attack surface reports

- Enhanced logic to detect and handle empty or invalid LLM-generated vulnerability reports and attack surface descriptions.
- Prevented saving or updating database records with empty LLM report content for vulnerabilities and subdomains.
- Improved update logic to only overwrite fields with non-empty LLM responses.
- Added normalization and validation for list-like empty values in vulnerability reports and attack surface data.

psyray committed:
feat(llm): add delete vulnerability analysis, show LLM model, and improve modal actions

- Added API and UI support for deleting LLM-generated vulnerability analyses.
- Enhanced vulnerability detail modals to display the LLM model used for generation.
- Improved modal UI with new action buttons for regenerating and deleting analyses, including confirmation dialogs and error handling.
- Updated backend to tag and extract LLM model information in vulnerability reports.

psyray committed:
fix: apply sourcery recommendations

psyray committed:
refactor(llm): centralise and reuse LLM report and text emptiness checks

- Removed unnused JS function
- Centralised and refactored utility functions for checking empty LLM report fields, attack surfaces, and text values in llm/utils.py.
- Updated backend logic to use the new shared utility functions for emptiness checks in LLM vulnerability report generation and attack surface suggestion APIs.
- Improved code maintainability by eliminating duplicate logic and ensuring consistent emptiness checks across the codebase.

psyray committed:
fix: replace html with text to prevent html injection

psyray committed:
refactor(llm): remove unused LLM model validation function from utils

- Deleted the validate_llm_model function from the LLM utils module, simplifying the codebase by removing unused or redundant logic.

psyray commented:
Copilot comments has been fixed, tests OK, can't be merged for me

AnonymousWP commented:
> Copilot comments has been fixed, tests OK, can't be merged for me

Can't or can?

psyray committed:
fix: improve XSS protection and error handling in LLM vulnerability modal and API

- Enhanced XSS protection in the LLM vulnerability modal by using safer DOM manipulation and sanitization methods.
- Improved error handling and logging when fetching the default LLM model in the API, ensuring robustness if the model is unavailable.
- Added a safeguard when assigning the LLM model name in the vulnerability report task to prevent attribute errors.

psyray commented:
> > Copilot comments has been fixed, tests OK, can't be merged for me
> 
> Can't or can?

Can, sorry :grin: 
I've applied CoPilot recommendations
Ready to merge

AnonymousWP commented:
Just curious if Copilot has nothing to nag about anymore, so I'm re-requesting. 😉

psyray commented on web/static/custom/custom.js:
```diff
@@ -3110,10 +3275,10 @@ async function show_attack_surface_modal(endpoint_url, id) {
     try {
         // Fetch models from the unified endpoint that combines GPT and Ollama models
-        const response = await fetch('/api/tools/llm_models');
+        const response = await fetch('/api/tools/llm_models/');
```
call without slash make a 302 redirect

psyray committed:
fix: prevent HTML content injection

psyray committed:
fix(llm): improve empty text detection and enhance LLM config dialog interactions

- Enhanced the logic for detecting empty or null-like text values in LLM utilities.
- Improved the LLM configuration choice dialog by replacing inline JavaScript with safer event binding and adding button IDs for better interaction handling.
- Refined the logic for determining GPT model selection and OpenAI key presence in the LLM vulnerability report API.
- Improved caching logic and code clarity in the LLM vulnerability report task.

sundayz-hunter reviewed (approved):
Looks good to me !
I’ll take a look separately to see if I can fix some of the CodeQL alerts related to the API.
